plugin/pkg/admission/storageclass/setdefault/admission.go

```diff
@@ -77,17 +77,61 @@ func (a *claimDefaulterPlugin) Validate() error {
 	return nil
 }
 
-// Admit sets the default value of a PersistentVolumeClaim's storage class, in case the user did
-// not provide a value.
+// Admit does two things:
+// 1. sets the default value of a PersistentVolumeClaim's storage class, in case
+// the user did not provide a value.
 //
+// 2. prevents from having two default storage classes.
+func (c *claimDefaulterPlugin) Admit(a admission.Attributes) error {
+	if a.GetResource().GroupResource() == api.Resource("persistentvolumeclaims") {
+		return c.admitPeristentVolumeClaim(a)
+	}
+	if a.GetResource().GroupResource() == storage.Resource("storageclasses") {
+		return c.admitStorageClass(a)
+	}
+	return nil
+}
+
+func (c *claimDefaulterPlugin) admitStorageClass(a admission.Attributes) error {
+	if len(a.GetSubresource()) != 0 {
+		return nil
+	}
+	class, ok := a.GetObject().(*storage.StorageClass)
+	// if we can't convert then we don't handle this object so just return
+	if !ok {
+		return nil
+	}
+
+	if !storageutil.IsDefaultAnnotation(class.ObjectMeta) {
+		// this class is not default
+		return nil
+	}
+
+	def, err := getDefaultClass(c.lister)
```

Comment:
Can we change this variable name to something more readable. like `defaultClass` ?